src/test/java/com/icthh/xm/gate/security/oauth2/IdpAuthenticationSuccessHandlerUnitTest.java

```diff
@@ -0,0 +1,240 @@
+
+    @BeforeEach
+    void setUp() {
+        tenantContextHolder.getPrivilegedContext().destroyCurrentContext();
+    }
+
+    @Test
+    public void test_shouldFailOnStatefulMode() throws IOException {
+
+        String tenantKey = "tenant1";
+        String clientKeyPrefix = "Auth0_";
+
+        int clientsAmount = 1;
+
+        IdpPublicConfig idpPublicConfig = registerPublicConfigs(clientKeyPrefix, tenantKey, clientsAmount, true, true, true);
+        IdpPrivateConfig idpPrivateConfig = registerPrivateConfigs(clientKeyPrefix, tenantKey, clientsAmount, true, true);
+
+        validateRegistration(tenantKey, clientKeyPrefix, clientsAmount, idpPublicConfig, idpPrivateConfig);
+
+        TenantContextUtils.setTenant(tenantContextHolder, tenantKey);
+        jhipsterProperties.getSecurity().getClientAuthorization().setAccessTokenUri("http://uaa.com");
+
+        this.idpAuthenticationSuccessHandler =
+            new IdpAuthenticationSuccessHandler(objectMapper, restTemplate, tenantContextHolder, idpConfigRepository, jhipsterProperties);
+
+        Exception exception = null;
+        try {
+            idpAuthenticationSuccessHandler.onAuthenticationSuccess(null, null, null);
+        } catch (Exception t) {
+            exception = t;
+        }
+
+        assertNotNull(exception);
+        assertEquals("Stateful mode not supported yet", exception.getMessage());
+
+    }
+
+    @SuppressWarnings("unchecked")
+    @Test
+    public void test_shouldSuccessfullyReturnData() throws IOException {
+
+        String tenantKey = "tenant1";
+        String clientKeyPrefix = "Auth0_";
+
+        int clientsAmount = 1;
+
+        IdpPublicConfig idpPublicConfig = registerPublicConfigs(clientKeyPrefix, tenantKey, clientsAmount, true, true, false);
+        IdpPrivateConfig idpPrivateConfig = registerPrivateConfigs(clientKeyPrefix, tenantKey, clientsAmount, true, true);
+
+        validateRegistration(tenantKey, clientKeyPrefix, clientsAmount, idpPublicConfig, idpPrivateConfig);
+
+        TenantContextUtils.setTenant(tenantContextHolder, tenantKey);
+        jhipsterProperties.getSecurity().getClientAuthorization().setAccessTokenUri("http://uaa.com");
+
+        when(restTemplate.exchange(
+            anyString(),
+            eq(HttpMethod.POST),
+            any(),
+            (ParameterizedTypeReference<Map<String, Object>>) any()))
+            .thenReturn(buildResponse(true));
+
+        this.idpAuthenticationSuccessHandler =
+            new IdpAuthenticationSuccessHandler(objectMapper, restTemplate, tenantContextHolder, idpConfigRepository, jhipsterProperties);
+
+        MockHttpServletResponse response = new MockHttpServletResponse();
+        idpAuthenticationSuccessHandler.onAuthenticationSuccess(null, response, buildAuthentication("email"));
+
+        //Capture and validate args
+        ArgumentCaptor<String> accessTokenUriCaptor = ArgumentCaptor.forClass(String.class);
```

Comment:
replace argument captor to eq or refEq validation

Reply:
Remark applied